performance/odm/playlist.xml

```diff
@@ -16,8 +16,10 @@
 	xsi:noNamespaceSchemaLocation="../../TestConfig/playlist.xsd">
 	<test>
 		<testCaseName>ODM</testCaseName>
-		<command>chmod a+x ./scripts/benchmark.sh; \
-		./scripts/benchmark.sh $(TEST_RESROOT); \
+		<command>cd $(TEST_RESROOT)/ilog_wodm881; \
+		cp $(TEST_RESROOT)/../../../openjdk-tests/performance/odm/scripts/benchmark.sh .; \
```

Comment:
I am not sure the copy is necessary.

Reply:
Copying of this file is needed since it uses the current dir for setting the working dir (i.e. `WORKDIR`). That working dir var is used in the ODM script. We usually put all our scripts under the benchmark package folder so that everything is under one place and we don't have to worry about paths that much.

Reply:
- TKG will change the dir into `$(REPORTDIR)`, so that all test output will be archived if there is any failure. If we `cd` into another dir, test output will not be archived. If we really have to change dir, then we need to copy all scripts into `$(REPORTDIR)`
- I still do not really get why we cannot pass in `$(TEST_RESROOT)`. The original script worked before.

Reply:
@llxia I see that `openjdk-tests/performance/odm/autoGen.mk` creates a dir (i.e. something like this `/java/jenkins/workspace/Piyush_ODM_x86-64_linux/openjdk-tests/TestConfig/test_output_15436147118144/ODM_0`). I believe that's the `$(REPORTDIR)` dir? So that `autoGen.mk` changes the dir to that before calling `openjdk-tests/performance/odm/playlist.xml`. Without copying this `benchmark.sh` file, I was getting this error `chmod: cannot access './scripts/benchmark.sh': No such file or directory` since `benchmark.sh` is in `openjdk-tests/performance/odm/scripts/` and not `test_output_15436147118144/ODM_0`. I believe you should have got the same error. Anyway, what test output are we archiving if there's a failure? I believe we want to archive java core/dump files if those files are generated in that directory.

Reply:
A more general Q: What's the idea behind using `$(REPORTDIR)` (i.e. `/java/jenkins/workspace/Piyush_ODM_x86-64_linux/openjdk-tests/TestConfig/test_output_15436147118144/ODM_0`)? I believe TKG will be creating a new dir for each iteration (i.e. test_output_15436147118144/ODM_0, ODM_1, ODM_2...`) if we decide to run it with multiple iterations. Since we're are downloading the benchmark package here `${BUILD_ROOT}/performance/odm` (i.e. `/java/jenkins/workspace/Piyush_ODM_x86-64_linux/jvmtest/performance/odm`), we'll be running the java command from one of the scripts (i.e. `run_ilog_with_gcmv.sh`) in that directory. I'm planning to remove `run_ilog_with_gcmv.sh` from here in order to avoid having duplicate scripts and add any minor changes to our ODM perf repo.

Reply:
> Without copying this `benchmark.sh` file, I was getting this error `chmod: cannot access './scripts/benchmark.sh': No such file or directory` 


I think $(REPORTDIR) was introduced after ODM was added, so we have some path issue that needs to be fixed.


> What's the idea behind using $(REPORTDIR)

The idea is to archive test output (including core/dump files) if there is a failure (without any user intervention). Each test has its own $(REPORTDIR). The build will automatically archive test_output_XXX (parent dir of $(REPORTDIR)) if the build is unstable (test failure) for debugging purpose.

In the existing code, ODM may have two archives:
- $(REPORTDIR) if the test fails
- `jvmtest/performance/odm/ilog_wodm881/leftoverResults` regardless of the test results ===> In order to compare measurements in ODM test, we also keep output files regardless of the test results. Ideally all output should be in $(REPORTDIR). We can refine this later if needed.

If we `cd` into another dir, core/dump files will not be archived in failure case. And to avoid cp test material around, I would prefer to use `$(TEST_RESROOT)`.

Reply:
@llxia All our benchmark scripts assume that we're inside the benchmark package directory while running the benchmark scripts. For example, if we aren't inside the `ilog_wodm881` dir (i.e. inside $(REPORTDIR)), `run_ilog_with_gcmv.sh` will throw this error `grep: bin/ra.xml: No such file or directory` since it assumes that it's in the same level dir as `bin`. Also, all our benchmark scripts take care of managing the dump/core files. Currently, for ODM, we're using `PERFFARM_DUMP="${WORKSPACE}/dump` to put dumps there but I'll change that to put it somewhere else (maybe `ilog_wodm881/leftoverResults`). Btw I'll be removing `run_ilog_with_gcmv.sh` from this repo in another PR in order to avoid having duplicates since we keep that script along with other ODM material. I'll make the necessary changes to manage the dump in our original ODM script so I won't make any changes to `run_ilog_with_gcmv.sh` in this repo for now. So for now, we'll be fine changing the dir from $(REPORTDIR) to `$(TEST_RESROOT)/ilog_wodm881` if you're okay with it. Please let me know if you have any other concerns that might be stopping us from merging this.